docs/fsharp/whats-new/fsharp-10.md

```diff
@@ -0,0 +1,382 @@
+
+## Typed bindings in computation expressions without parentheses
+
+F# 10 removes the requirement for parentheses when adding type annotations to computation expression bindings.
+You can now add type annotations on `let!`, `use!`, and `and!` bindings using the same syntax as ordinary `let` bindings.
+
+Previously, you had to use parentheses for type annotations:
+
+```fsharp
+async {
+    let! (a: int) = fetchA()
+    and! (b: int) = fetchB()
+    use! (d: MyDisposable) = acquireAsync()
+    return a + b
+}
+```
+
+In F# 10, you can write type annotations without parentheses:
+
+```fsharp
+async {
+    let! a: int = fetchA()
+    and! b: int = fetchB()
+    use! d: MyDisposable = acquireAsync()
+    return a + b
+}
+```
+
+## Allow `_` in `use!` bindings
+
+You can now use the discard pattern (`_`) in `use!` bindings within computation expressions.
+This aligns the behavior of `use!` with regular `use` bindings.
+
+Previously, the compiler rejected the discard pattern in `use!` bindings, forcing you to create throwaway identifiers:
+
+```fsharp
+counterDisposable {
+    use! _ignored = new Disposable()
+    // logic
+}
+```
+
+In F# 10, you can use the discard pattern directly:
+
+```fsharp
+counterDisposable {
+    use! _ = new Disposable()
+    // logic
+}
+```
+
+This clarifies intent when binding asynchronous resources whose values are only needed for lifetime management.
+
+## Rejecting pseudo-nested modules in types
+
+The compiler now raises an error when you place a `module` declaration indented at the same structural level inside a type definition.
+This tightens structural validation to reject misleading module placement within types.
+
+Previously, the compiler accepted `module` declarations indented within type definitions, but it actually created these modules as siblings to the type rather than nesting them within it:
+
+```fsharp
+type U =
+    | A
+    | B
+    module M = // Silently created a sibling module, not nested
+        let f () = ()
+```
+
+With F# 10, this pattern raises error FS0058, forcing you to clarify your intent with proper module placement:
+
+```fsharp
+type U =
+    | A
+    | B
+
+module M =
+    let f () = ()
+```
+
+## Deprecation warning for omitted `seq`
+
+The compiler now warns you about bare sequence expressions that omit the `seq` builder.
+When you use bare range braces like `{ 1..10 }`, you'll see a deprecation warning encouraging you to use the explicit `seq { ... }` form.
+
+Historically, F# allowed a special-case "sequence comprehension lite" syntax where you could omit the `seq` keyword:
+
+```fsharp
+{ 1..10 } |> List.ofSeq  // implicit sequence
+```
+
+In F# 10, the compiler warns about this pattern and encourages the explicit form:
+
+```fsharp
+seq { 1..10 } |> List.ofSeq
+```
+
+This is currently a warning, not an error, giving you time to update your codebase.
```

Comment:
This should IMO say the exact warning code and maybe a small `no warn` snippet so that is super clear how to keep current behavior easily.